lib/spack/env/cc

```diff
@@ -51,6 +51,9 @@ parameters=(
     SPACK_CXX_RPATH_ARG
     SPACK_F77_RPATH_ARG
     SPACK_FC_RPATH_ARG
+    SPACK_DTAGS_TO_ENABLE
+    SPACK_DTAGS_TO_DISABLE
```

Comment:
This feels very specific to me, and it feels kludgy to have two variables...  Could we generalize this?  it actually seems like we could do it more cleanly by just adding either `--enable-new-dtags`/`--disable-new-dtags` to `SPACK_LDFLAGS` in the build environment, then a single more general option like `SPACK_STRIP_LDFLAGS`, which would strip a set of flags from the link line - not just this one.  That mechanism would be a similar amount of code, but there are other places where it would be useful -- For example, I've seen users want to do things like strip fPIC for certain compilers on certain platforms, and we could (in future PRs) allow packages, compilers, or platforms to set some of these things to ease porting.  What do you think?